http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -677,13 +695,18 @@ private void handleRouteMatch(
         io.netty.handler.codec.http.HttpRequest nativeRequest = request.getNativeRequest();
         if (!route.isExecutable() && io.micronaut.http.HttpMethod.permitsRequestBody(request.getMethod()) && nativeRequest instanceof StreamedHttpRequest) {
             Optional<MediaType> contentType = request.getContentType();
-            HttpContentProcessor<?> processor = contentType
-                .flatMap(type ->
-                    beanLocator.findBean(HttpContentSubscriberFactory.class,
-                        new ConsumesMediaTypeQualifier<>(type))
-                ).map(factory ->
-                    factory.build(request)
-                ).orElse(new DefaultHttpContentProcessor(request, serverConfiguration));
+            Optional<Class<?>> bodyType = route.getBodyArgument().map(Argument::getType);
+            boolean isFunctionalBean = route.getAnnotation("io.micronaut.function.FunctionBean") != null;
+
+            HttpContentProcessor<?> processor;
+            if (isFunctionalBean && bodyType.isPresent() && String.class.isAssignableFrom(bodyType.get())) {
```

Comment:
Why was this change necessary? Seems like a hack to work around another problem

Reply:
The current behaviour of how request deserialization works for FunctionBean is `really` inconsistent.
1. The documentation says that parameter should contain the entire body . 
> In addition, functions have an input and/or an output. The input is represented by the accepted argument and represents the body consumed by the function and the output is represented by the return value of the function. The input and the output should be either a Java primitive or simple type (int, String etc.) or a POJO.

2. For groovy based FunctionBean and with json request we bind json parameter to bean parameter

3. For java based FunctionBean binding of json fields does not work at all 

4. For xml request we have a test that also binding entire body to the parameter https://github.com/micronaut-projects/micronaut-core/blob/afc3fb377fea37f942c1a251be19b58f9f989b22/function-web/src/test/groovy/io/micronaut/function/web/JavaLambdaFunctionSpec.groovy#L113

I struggle to fix the tests, but due to these inconsistencies once I fix one of them, another gets broken.

Reply:
Yeah so I wouldn't hide any inconsistencies we have with a workaround. Let the tests be failing so we can have a clear understanding of what isn't working and then we can discuss how it should behave. Thoughts on the above @graemerocher ? I really haven't worked with the functions enough to determine how they should work.